packages/core-data/src/selectors.js

```diff
@@ -730,5 +730,14 @@ export function __experimentalGetTemplateForLink( state, link ) {
 		'find-template': link,
 	} );
 
-	return records?.length ? records[ 0 ] : null;
+	const template = records?.length ? records[ 0 ] : null;
+	if ( template ) {
```

Comment:
I don't understand the change here? Is it to make sure we retrieve the "edited" version and not the "persisted" one? What's the difference?

Reply:
The difference is the title object getEditedEntityRecord always returns title as a string while getEntityRecords returns title as an object with raw and rendered properties. Also, it seems more correct to retrieve the edit version in case there are edits.